src/main/scala/org/broadinstitute/dsde/workbench/avram/dataaccess/HttpSamDao.scala

```diff
@@ -2,31 +2,25 @@ package org.broadinstitute.dsde.workbench.avram.dataaccess
 
 import java.util.logging.Logger
 
-import com.softwaremill.sttp._
-import io.circe.Decoder.Result
+import mouse.all._
+import cats.implicits._
 import io.circe.{DecodingFailure, Json, ParsingFailure}
 import io.circe.generic.auto._
 import io.circe.parser._
-import org.broadinstitute.dsde.workbench.avram.util.{ErrorResponse, RestClient}
+import org.broadinstitute.dsde.workbench.avram.util.transformers._
+import org.broadinstitute.dsde.workbench.avram.util.{AvramError, RestClient}
 
 class HttpSamDao(samUrl: String) extends SamDao with RestClient {
   private val log: Logger = Logger.getLogger(getClass.getName)
 
-  override def getUserStatus(token: String): Either[ErrorResponse, SamUserInfoResponse] = {
+  override def getUserStatus(token: String): AvramResult[SamUserInfoResponse] = {
     val request = buildAuthenticatedGetRequest(samUrl, "/register/user/v2/self/info", token)
-    val response: Id[Response[String]] = request.send()
-    response.body match {
-      case Left(error: String) => Left(ErrorResponse(response.code, error))
-      case Right(content: String) =>
-        parse(content) match {
-          case Left(error: ParsingFailure) => Left(ErrorResponse(500, error.message))
-          case Right(json: Json) =>
-            val result: Result[SamUserInfoResponse] = json.as[SamUserInfoResponse]
-            result match {
-              case Left(error: DecodingFailure) => Left(ErrorResponse(500, error.message))
-              case Right(userInfo: SamUserInfoResponse) => Right(userInfo)
-            }
-        }
-    }
+    for {
+      response <- request.send()                        |> ioToResult
```

Comment:
Illustrates the user of utility methods to "lift" types into `AvramResult`.

Note: `|>` (or `thrush` alternatively) is just function application. E.g. `foo(3)` is equivalent to `3 |> foo`. This is just syntax, not essential for core functionality.

I thought this format was a little easier to read than `ioToResult(request.send())`. I like how the code itself is clean and the types are on the right almost like documentation.

Reply:
Interesting. There is definitely some appeal to how it reads. I'd love to not have to do an explicit conversion here but I guess there isn't any way around that. Since I did not know this syntax, I was going to write a bunch of functions, e.g. (types omitted because, frankly, I'd need IntelliJ to tell me what they are):
```
def sendRequest(request): = {
  response = request.send()
  ioToResult(response)
}
```
OTOH, that's a bit of extra boilerplate code. I might prefer the manual formatting chore over having to write all of those functions.

Side note related to slack discussions about auto-formatting: I wonder if scalafmt would respect the desire for the `|>`s to line up vertically.

Reply:
It _might_ be possible to make `ioToResult` an implicit conversion so you wouldn't need to specify it at all. I'm wary however of implicit conversions because it can make the code feel too magical. Maybe I'll see if that even works..